simulator/tools/rofi_hal.hpp

```diff
@@ -0,0 +1,51 @@
+#include <memory>
+
+namespace rofi
+{
+    namespace hal
+    {
+        class RoFI
+        {
+        public:
+            class Data;
+            class Joint;
+
+        private:
+            boost::shared_ptr< Data > rdata;
+
+        public:
+            // TODO constructors
+            explicit RoFI();
+
+            Joint getJoint( int index );
+
+            class Joint
+            {
+            public:
+                class Data;
+                // TODO class JointError {};
+
+                Data * jdata;
```

Comment:
Reference or pointer?

Reply:
For `class Joint` to be copy assignable, there has to be a pointer.
I could use a reference wrapper or ensure in implementation, that the pointer never goes to null.

Reply:
Fair point. Leave it as it is.